plugins/account_history/plugin.cpp

```diff
@@ -135,242 +157,290 @@ if( options.count(name) ) { \
         }
 
         void operator()(const feed_publish_operation& op) {
-            impacted.insert(op.publisher);
+            impacted.insert(make_pair(op.publisher, operation_direction_type::any));
         }
 
         void operator()(const limit_order_create_operation& op) {
-            impacted.insert(op.owner);
+            impacted.insert(make_pair(op.owner, operation_direction_type::any));
         }
 
         void operator()(const fill_order_operation& op) {
-            impacted.insert(op.current_owner);
-            impacted.insert(op.open_owner);
+            impacted.insert(make_pair(op.current_owner, operation_direction_type::sender));
+            impacted.insert(make_pair(op.open_owner, operation_direction_type::receiver));
         }
 
         void operator()(const limit_order_cancel_operation& op) {
-            impacted.insert(op.owner);
+            impacted.insert(make_pair(op.owner, operation_direction_type::any));
         }
 
         void operator()(const pow_operation& op) {
-            impacted.insert(op.worker_account);
+            impacted.insert(make_pair(op.worker_account, operation_direction_type::any));
         }
 
         void operator()(const fill_vesting_withdraw_operation& op) {
-            impacted.insert(op.from_account);
-            impacted.insert(op.to_account);
+            if (op.from_account == op.to_account) {
+                impacted.insert(make_pair(op.from_account, operation_direction_type::any));
+                impacted.insert(make_pair(op.to_account, operation_direction_type::any));
+            } else {
+                impacted.insert(make_pair(op.from_account, operation_direction_type::sender));
+                impacted.insert(make_pair(op.to_account, operation_direction_type::receiver));
+            }
         }
 
         void operator()(const shutdown_witness_operation& op) {
-            impacted.insert(op.owner);
+            impacted.insert(make_pair(op.owner, operation_direction_type::any));
         }
 
         void operator()(const custom_operation& op) {
             for (auto& s: op.required_auths) {
-                impacted.insert(s);
+                impacted.insert(make_pair(s, operation_direction_type::any));
             }
         }
 
         void operator()(const request_account_recovery_operation& op) {
-            impacted.insert(op.account_to_recover);
+            impacted.insert(make_pair(op.account_to_recover, operation_direction_type::any));
         }
 
         void operator()(const recover_account_operation& op) {
-            impacted.insert(op.account_to_recover);
+            impacted.insert(make_pair(op.account_to_recover, operation_direction_type::any));
         }
 
         void operator()(const change_recovery_account_operation& op) {
-            impacted.insert(op.account_to_recover);
+            impacted.insert(make_pair(op.account_to_recover, operation_direction_type::any));
```

Comment:
missing `new_recovery_account`:

```
struct change_recovery_account_operation : public base_operation {
            account_name_type account_to_recover;     ///< The account that would be recovered in case of compromise
            account_name_type new_recovery_account;   ///< The account that creates the recover request
```